tls/s2n_psk.h

```diff
@@ -89,9 +95,12 @@ int s2n_psk_calculate_binder(struct s2n_psk *psk, const struct s2n_blob *binder_
 int s2n_psk_verify_binder(struct s2n_connection *conn, struct s2n_psk *psk,
         const struct s2n_blob *partial_client_hello, struct s2n_blob *binder_to_verify);
 
-typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn, 
-                                          struct s2n_psk_identity *identities, size_t identities_length,
-                                          uint16_t *chosen_wire_index);                     
-/* This function will be labeled S2N_API and become a publicly visible api once we release the psk API. */
-int s2n_config_set_psk_selection_callback(struct s2n_connection *conn, s2n_psk_selection_callback cb);
+/* These functions and types will be publicly visible once we release the psk feature. */
 
+struct s2n_psk_identity_list;
+typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn,
+                                          struct s2n_psk_identity_list *psk_identity_list,
+                                          uint16_t *chosen_wire_index);
+int s2n_config_set_psk_selection_callback(struct s2n_config *config, s2n_psk_selection_callback cb);
+int s2n_psk_identity_list_next(struct s2n_psk_identity_list *list, uint8_t **next_identity, uint16_t *identity_size);
+int s2n_psk_identity_list_reset(struct s2n_psk_identity_list *list);
```

Comment:
Do we anticipate our customers using these helper functions? What is the reasoning behind releasing these functions to the public?

Reply:
Yes. Because s2n_psk_identity_list is opaque, these methods are the only way to interact with it. This is how callers get identities from the list.